crates/debuginfo/src/source_manager.rs

```diff
@@ -254,6 +254,14 @@ pub trait SourceManagerExt: SourceManager {
 #[cfg(feature = "std")]
 impl<T: ?Sized + SourceManager> SourceManagerExt for T {}
 
+/// [SourceManagerSync] is a [SourceManager] that can be used in a multi-threaded context.
+///
+/// This trait should not be implemented directly; there is a blanket implementation for all types
+/// that implement [SourceManager] + [Send] + [Sync].
```

Comment:
I think I would just phrase this as:

> [SourceManagerSync] is a marker trait for [SourceManager] implementations that are also `Send + Sync`, and is automatically implemented for any [SourceManager] that meets those requirements.
>
> [SourceManager] is a supertrait of [SourceManagerSync], so you may use instances of the latter where the former is required, either implicitly or via explicit downcasting, e.g. `Arc<dyn SourceManagerSync> as Arc<dyn SourceManager>`.

The compiler won't let you implement `SourceManagerSync` manually anyway, so it isn't actually necessary to instruct consumers not to implement it.